packages/core/src/extensions/Blocks/api/block.ts

```diff
@@ -243,122 +221,44 @@ export function wrapInBlockStructure<
   };
 }
 
-// A function to create custom block for API consumers
-// we want to hide the tiptap node from API consumers and provide a simpler API surface instead
-export function createBlockSpec<
-  BType extends string,
-  PSchema extends PropSchema,
-  ContainsInlineContent extends boolean,
-  BSchema extends BlockSchemaWithBlock<BType, PSchema, ContainsInlineContent>
->(
-  blockConfig: BlockConfig<BType, PSchema, ContainsInlineContent, BSchema>
-): BlockSpec<BType, PSchema, ContainsInlineContent> {
-  const node = createTipTapBlock<
-    BType,
-    ContainsInlineContent,
-    {
-      editor: BlockNoteEditor<BSchema>;
-      domAttributes?: BlockNoteDOMAttributes;
-    }
-  >({
-    name: blockConfig.type,
-    content: (blockConfig.containsInlineContent
-      ? "inline*"
-      : "") as ContainsInlineContent extends true ? "inline*" : "",
-    selectable: true,
-
-    addAttributes() {
-      return propsToAttributes(blockConfig);
-    },
-
-    parseHTML() {
-      return parse(blockConfig);
-    },
-
-    addNodeView() {
-      return ({ getPos }) => {
-        // Gets the BlockNote editor instance
-        const editor = this.options.editor;
-        // Gets the block
-        const block = getBlockFromPos<
-          BType,
-          PSchema,
-          ContainsInlineContent,
-          BSchema
-        >(getPos, editor, this.editor, blockConfig.type);
-        // Gets the custom HTML attributes for `blockContent` nodes
-        const blockContentDOMAttributes =
-          this.options.domAttributes?.blockContent || {};
-
-        const output = blockConfig.render(block, editor);
-
-        return wrapInBlockStructure<BType, PSchema>(
-          output,
-          block.type,
-          block.props,
-          blockConfig.propSchema,
-          blockContentDOMAttributes
-        );
-      };
-    },
-  });
+// Helper type to keep track of the `name` and `content` properties after calling Node.create.
+type StronglyTypedTipTapNode<
+  Name extends string,
+  Content extends "inline*" | "table" | "tableRow+" | ""
+> = Node & { name: Name; content: Content };
+
+export function createStronglyTypedTiptapNode<
+  Name extends string,
+  Content extends "inline*" | "table" | "tableRow+" | ""
+>(config: NodeConfig & { name: Name; content: Content }) {
+  return Node.create(config) as StronglyTypedTipTapNode<Name, Content>; // force re-typing (should be safe as it's type-checked from the config)
+}
 
+export function createInternalBlockSpec<T extends BlockConfig>(
```

Comment:
This is a really weird function, it just returns the same object? Seems a bit hacky

Reply:
true, it's a bit hacky. Added comment:

```
// This helper function helps to instantiate a blockspec with a
// config and implementation that conform to the type of Config
```

I couldn't get it working without (maybe it's possible with `satisfies` somewhere, this makes it easy to instantiate a blockImplementation with strongly typed render / externalhtml functions